docs/GUI_PLUGINS_ARCHITECTURE.md

```diff
@@ -0,0 +1,294 @@
+# Edge React GUI - Fiat Plugin Architecture
+
+## Overview
+
+The Edge wallet implements a sophisticated plugin architecture for integrating fiat on/off ramp providers. This system enables users to buy and sell cryptocurrency through various payment methods and providers while maintaining a consistent user experience across different regions and currencies.
+
+## Architecture Components
+
+### 1. Directory Structure
+
+```
+src/plugins/gui/
+├── fiatPlugin.ts                    # Core plugin framework
+├── fiatPluginTypes.ts              # Type definitions
+├── fiatProviderTypes.ts            # Provider-specific types
+├── providers/                      # Individual provider implementations
+│   ├── paybisProvider.ts          # Paybis implementation
+│   ├── banxaProvider.ts           # Banxa implementation
+│   ├── moonpayProvider.ts         # Moonpay implementation
+│   └── ...                        # Other providers
+├── scenes/                         # Plugin UI scenes
+│   ├── FiatPluginWebView.tsx      # WebView wrapper
+│   ├── AddressFormScene.tsx       # Address collection
+│   └── InfoDisplayScene.tsx       # Information display
+└── util/
+    └── providerUtils.ts           # Shared utilities
+
+src/constants/plugins/
+├── sellPluginList.json            # Sell payment method configurations
+├── sellPluginListOverride.json    # Environment-specific overrides
+├── buyPluginList.json             # Buy payment method configurations
+└── GuiPlugins.ts                  # Plugin registry
+```
+
+### 2. Core Type System
+
+#### Payment Types
+Defined in `fiatPluginTypes.ts`:
+
+```typescript
+export const asFiatPaymentType = asValue(
+  'ach',           // ACH bank transfer
+  'applepay',      // Apple Pay
+  'credit',        // Credit/debit card
+  'directtobank',  // Direct bank transfer
+  'iach',          // Instant ACH
+  'interac',       # Interac e-Transfer (Canada)
```

Comment:
//

Reply:
[insert that meme that guy with a mullet mouthing wtf]